Extensions/Notification/notificationtools.js

```diff
@@ -1,89 +1,84 @@
 /**
  * @memberof gdjs.evtTools
- * @class notification
+ * @namespace
  * @static
- * @private
+ * @public
  */
 
 //variables
 gdjs.evtTools.notification = {
+  status: 'default',
   notif: [],        //used for actions
-  event: [],        //used for conditions (typicply eventhandlers)
+  event: [],        //used for conditions (typically eventhandlers)
 };
 
 gdjs.evtTools.notification.requestPermission = function() {
-  //Check if the user has already granted permission
-  if (Notification.permission === 'granted') {
-    return true;
-  }
-
-  //If they haven't ask for it
-  else if (Notification.permission !== 'denied') {
     Notification.requestPermission().then(function(status) {
-      if (status === "granted") {
-          return true;
-      }
+      gdjs.evtTools.notification.status = status;
     });
-  }
+}
 
-  //The user does not grant permission or does not respond to the prompt
-  else {
-    return false;
+gdjs.evtTools.notification.permissionGranted = function() {
+  if (Notification.permission === 'granted') {
+    return true
+  } else {
+    return false
   }
-};
+}
 
 gdjs.evtTools.notification.isSupported = function() {
   //Check if the window object contains a Notification
-  if (!('Notification' in window)) {
-    return false;
-  } 
-  
-  //The broswer is outdated and does not support notifications
-  else {
-    return true;
-  }
+  if ('Notification' in window) return true
 }
 
-gdjs.evtTools.notification.createNotification = function(id, title, msg, img, lng, tag, silence, vibe) {
+gdjs.evtTools.notification.createNotification = function(id, title, body, icon, lang, requireInteraction, silent, vibrate) {
   //Notification configuration see https://developer.mozilla.org/en-US/docs/Web/API/Notification for more information
   const config = {
-    body: msg,
+    body: body,
     data: id,
-    icon: img,
-    lang: lng,
-    tag: tag,
```

Comment:
Instead of fully removing tag you can use the ID as tag

Reply:
Do you mean swapping the id for the tag value like so ``tag: id``?

Reply:
Yes. As you can see on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Notification/Notification), the tag option is meant as an ID while the data option is meant as arbitrary data.